linkerd/core/src/main/scala/io/buoyant/linkerd/RequestActiveTracer.scala

```diff
@@ -0,0 +1,263 @@
+        formatDelegation(remainingTree, nodes :+ PathNode(path.show, name, t), clientName)
+      case d@Delegate(path, dentry, remainingTree) =>
+        formatDelegation(remainingTree, nodes :+ PathNode(path.show, dentry.show, d), clientName)
+      case u@Union(path, dentry, remainingWeightedTrees@_*) =>
+        remainingWeightedTrees.map { wd =>
+          formatDelegation(wd.tree, nodes :+ PathNode(path.show, dentry.show, u), clientName)
+        }.find(!_.isEmpty).toList.flatten
+      case a@Alt(path, dentry, remainingTrees@_*) =>
+        remainingTrees.map { d =>
+          formatDelegation(d, nodes :+ PathNode(path.show, dentry.show, a), clientName)
+        }.find(!_.isEmpty).toList.flatten
+      case _ => List.empty
+    }
+  }
+
+  private def getRequestTraceResponse(resp: Response, stopwatch: Stopwatch.Elapsed) = {
+
+    val EmptyDelegateTree = DelegateTree.Empty(Path.empty, Dentry.nop)
+
+    val serviceName = DstPathCtx.current match {
+      case Some(dstPath) => dstPath.path
+      case None => Path.empty
+    }
+
+    val selectedEndpoint = endpoint.addr match {
+      case inetAddr: Address.Inet => inetAddr.addr.toString.stripPrefix("/")
+      case _ => ""
+    }
+
+    val clientPath = DstBoundCtx.current match {
+      case None => Path.empty
+      case Some(addrSet) => addrSet.name.id match {
+        case p: Path => p
+        case _ => Path.empty
+      }
+    }
+
+    val lbSet = DstBoundCtx.current match {
+      case None => Future.value(Addr.Neg)
+      case Some(addrSet) => addrSet.name.addr.changes.toFuture
+    }
+
+    val addresses = lbSet.map {
+      case Addr.Bound(a, _) => Some(
+        a.map {
+          case inetAddr: Address.Inet => inetAddr.addr.toString.stripPrefix("/")
+          case _ => ""
+        }
+      )
+      case _ => None
+    }
+
+    val dtreeF = namers.interpreter match {
+      case delegator: Delegator => delegator.delegate(dtab.dtab(), serviceName)
+        .map(Some(_))
+      case _ => Future.None
+    }
+
+    dtreeF.joinWith(addresses) {
+      case (dTree: Option[DelegateTree[Name.Bound]], addrSet: Option[Set[String]]) =>
+        val tree = formatDelegation(dTree.getOrElse(EmptyDelegateTree), List.empty, clientPath)
+
+        resp.contentString = formatRouterContext(
+          resp.contentString,
+          serviceName.show,
+          clientPath.show,
+          selectedEndpoint,
+          addrSet,
+          tree.map(_.toString),
+          stopwatch().inMillis.toString
+        )
+        //We set the content length of the response in order to view the content string in the response
+        // entirely
+        resp.contentLength = resp.content.length
+        resp
+    }
+  }
+
+  override def apply(
+    req: Request,
+    svc: Service[Request, Response]
+  ): Future[Response] = {
+
+    val maxForwards = Try(req.headerMap.get(MaxForwards).map(_.toInt))
+    val isAddRouterCtx = req.headerMap.get(AddRouterContextHeader).map(StringUtil.toBoolean).getOrElse(false)
```

Comment:
TIOLI I used to use this `opt.map(f).getOrElse(default)` pattern a lot but more recently I'm starting to favor

```
opt match {
  case Some(x) => f(x)
  case None => default
}
```

They're identical in effect but I find the second more readable.  Either way is fine, just wanted to point out the alternative.